lxd/util/net.go

```diff
@@ -158,6 +159,11 @@ func IsAddressCovered(address1, address2 string) bool {
 	address1 = CanonicalNetworkAddress(address1)
 	address2 = CanonicalNetworkAddress(address2)
 
+	// Do not allow empty addresses, as these would be unset configuration values.
+	if strings.HasPrefix(address1, ":") || strings.HasPrefix(address2, ":") {
+		return false
+	}
+
```

Comment:
I'm not sure I understand that part.

If I have `core.https_address` set to `:8443` it will cover `cluster.https_address` set to `1.2.3.4:8443`.

The change above suggests that it would now be considered as not covered and so would incorrectly try to start a second listener and fail.

Reply:
So what exactly is the functional difference between an unset `core.https_address` and an implicit wildcard address listening on the default port?

This check here omitted empty strings from being assigned ports, and the commit message mentions something about unsetting the address, which made me think we don't want to accept empty strings:
https://github.com/lxc/lxd/commit/ca3c94f04fe9ac52f5abf7518d9329006028324c

Reply:
An unset `core.https_address` means don't listen on anything at all, `:8443` means listen on every address and protocol.
In general, I'd expect `address` to actually refer to address:port so we can distinguish `:8443` from ``.

Reply:
I see now, thanks. I don't really like that 'CanonicalNetworkAddress' and by extension 'IsAddressCovered' basically require an 'if address != ""` before each call. I think returning an error from 'CanonicalNetworkAddress' if it receives an empty string might be a helpful reminder.